docs/docs/how-to/create_profile.md

```diff
@@ -250,4 +252,106 @@ filter the output using `jq`. For example, to get all rules that pertain to the
 run the following command:
 ```bash
 minder profile status list --name stacklok-remediate-profile -d -ojson 2>/dev/null | jq  -C '.ruleEvaluationStatus | map(select(.entityInfo.repo_name == "minder" and .status == "failure"))'
-```
\ No newline at end of file
+```
+
+## Defining Rule Names in Profiles
+
+In Minder profiles, rules are identified by their type and, optionally, a unique name.
+
+### Rule Types vs Rule Names
+
+Rule types are mandatory and refer to the kind of rule being applied. Rule names, on the other hand, are optional 
+identifiers that become crucial when multiple rules of the same type exist under an entity.
+
+```yaml
+repository:
+  - type: secret_scanning
+    name: "secret_scanning_github"
+    def:
+      enabled: true
+```
+In this example, `secret_scanning` is the rule type and `secret_scanning_github` is the rule name.
+
+### When are Rule Names Mandatory?
+
+If you're using multiple rules of the same type under an entity, each rule must have a unique name. This helps 
+distinguish between rules and understand their specific purpose.
+
+```yaml
+repository:
+  - type: secret_scanning
+    name: "secret_scanning_github"
+    def:
+      enabled: true
+  - type: secret_scanning
+    name: "secret_scanning_github_2"
+    def:
+      enabled: false
+```
+Here, we have two rules of the same type `secret_scanning` under the `repository` entity. Each rule has a unique name.
```

Comment:
The example from https://github.com/stacklok/minder-rules-and-profiles/blob/944f5f6cf60362250e82e874e4d7c296653f2ef9/profiles/github/profile.yaml#L38 might be more relevant:

````suggestion
```yaml
repository:
  - type: dependabot_configured
    name: go_dependabot
    def:
      package_ecosystem: gomod
      schedule_interval: weekly
      apply_if_file: go.mod
  - type: dependabot_configured
    name: node_dependabot
    def:
      package_ecosystem: npm
      schedule_interval: weekly
      apply_if_file: package.json
```
Here, we have two rules of the same type `dependabot_configured` under the `repository` entity. Each rule has a unique name.
````